includes/admin/class-wcs-admin-meta-boxes.php

```diff
@@ -84,9 +83,13 @@ public function add_meta_boxes() {
 
 		add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_subscription', 'normal', 'low' );
 
-		// Only display the meta box if an order relates to a subscription
-		if ( 'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID ) && wcs_order_contains_subscription( $post_ID, 'any' ) ) {
-			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_order', 'normal', 'low' );
+		// Get "Edit Order" screen ID, which differs if HPOS is enabled.
+		$screen         = wcs_is_custom_order_tables_usage_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
+		$current_screen = get_current_screen();
+
+		// Only display the meta box if viewing an order that contains a subscription.
+		if ( $current_screen->id === $screen && wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {
+			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', $screen, 'normal', 'low' );
```

Comment:
Added this `$current_screen->id === $screen` condition, as it was rendering on other screens (e.g. Edit Product).

Reply:
This condition use to be `'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID )` was that returning true on product pages?

Reply:
Before 59f50649ad81a3bca51a8740350d6529a46f79c2, I had the statement:

`if ( wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {`

This was returning throwing an error on product pages since I wasn't checking if it was the edit order screen (rookie WP hook user mistake).

`'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID )` was removed, since it was accessing `$post_ID` and I removed all references to `$post` or `$post_ID`.

Reply:
I need to double-check `wc_get_page_screen_id` will work with the earliest supported WC version.